[Rollup] migrate to rtl

kapral18 opened the pull request:
Closes #241531

## Summary
- Replace Enzyme-based Rollup Jest coverage with React Testing Library.
- Remove Enzyme page helper modules and switch to direct RTL interactions.
- Improve test isolation with jest.clearAllMocks() and async UI boundary waits.

## Test titles table of change
To make it easier to verify individual tests here is a table of test titles before and after change
for each changed file in the PR. 
- https://gist.github.com/kapral18/1202c2ea0ac138d273959f21a280b223

## Test plan
- [x] node scripts/jest x-pack/platform/plugins/private/rollup/

kapral18 committed:
test(rollup): migrate rollup tests to RTL

- Replace Enzyme-based Rollup Jest coverage with React Testing Library.
- Remove Enzyme page helper modules and switch to direct RTL interactions.
- Improve test isolation with jest.clearAllMocks() and async UI boundary waits.

Closes #241531

elasticmachine commented:
Pinging @elastic/kibana-management (Team:Kibana Management)

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/377552)
* Commit: 9a230f1b371b2de990575cbd89f6acca52a60287

### Failed CI Steps
* [FTR Configs #48](https://buildkite.com/elastic/kibana-pull-request/builds/377552#019b7f8f-1300-400a-8fc2-5419d5451f59)
* [Scout: [ platform / streams_app ] plugin](https://buildkite.com/elastic/kibana-pull-request/builds/377552#019b7f94-56b7-4a8c-ab07-8b842623e719)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/377552#019b7f8f-1300-400a-8fc2-5419d5451f59) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/377552/jobs/019b7f8f-1300-400a-8fc2-5419d5451f59/artifacts/019b7fc0-2a52-4757-a9b9-df4ad44aaeec) FTR Configs #<span></span>48 / Package policies Package Policy - upgrade when upgrading from an integration package to an input package where no required variable has been added upgrade upgrades the package policy and creates the correct templates

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :yellow_heart: [Build #377372](https://buildkite.com/elastic/kibana-pull-request/builds/377372) was flaky bab3e681a6ec79ec73eadf5ec23b1d40c560106b

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377552","number":377552,"commit":"9a230f1b371b2de990575cbd89f6acca52a60287"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377372","number":377372,"commit":"bab3e681a6ec79ec73eadf5ec23b1d40c560106b"}],"number":377552}
buildkite-pr-comment-->

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/377930)
* Commit: cfca8b888e49e7ebb59082822c2d72bbea6ff4d2

### Failed CI Steps
* [FTR Configs #57](https://buildkite.com/elastic/kibana-pull-request/builds/377930#019b8eb0-4618-42d2-ac86-319925dd08ea)
* [FTR Configs #58](https://buildkite.com/elastic/kibana-pull-request/builds/377930#019b8eb0-4619-48ef-ac1a-1f33d452e22f)
* [Scout: [ platform / streams_app ] plugin](https://buildkite.com/elastic/kibana-pull-request/builds/377930#019b8eb6-3b32-4f93-9e7f-b50869b70d2c)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/377930#019b8eb0-4619-48ef-ac1a-1f33d452e22f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/377930/jobs/019b8eb0-4619-48ef-ac1a-1f33d452e22f/artifacts/019b8ec6-4cbe-4ec0-9412-bea716688f6a) FTR Configs #<span></span>58 / Cloud Security Posture Test adding Cloud Security Posture Integrations CNVM CNVM AWS On Add Agent modal there should be modal that has Cloud Formation details as well as button that redirects user to Cloud formation page on AWS upon clicking them 
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/377930#019b8eb0-4618-42d2-ac86-319925dd08ea) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/377930/jobs/019b8eb0-4618-42d2-ac86-319925dd08ea/artifacts/019b8ed5-4115-4601-b8c9-a5e40015f7d2) FTR Configs #<span></span>57 / Fleet tasks Automatic agent upgrades should take agents on target version into account

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :yellow_heart: [Build #377814](https://buildkite.com/elastic/kibana-pull-request/builds/377814) was flaky aea8c221b9f7c32a991c23e6421b03be110d6f00
* :yellow_heart: [Build #377552](https://buildkite.com/elastic/kibana-pull-request/builds/377552) was flaky 9a230f1b371b2de990575cbd89f6acca52a60287
* :yellow_heart: [Build #377372](https://buildkite.com/elastic/kibana-pull-request/builds/377372) was flaky bab3e681a6ec79ec73eadf5ec23b1d40c560106b

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377930","number":377930,"commit":"cfca8b888e49e7ebb59082822c2d72bbea6ff4d2"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377814","number":377814,"commit":"aea8c221b9f7c32a991c23e6421b03be110d6f00"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377552","number":377552,"commit":"9a230f1b371b2de990575cbd89f6acca52a60287"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377372","number":377372,"commit":"bab3e681a6ec79ec73eadf5ec23b1d40c560106b"}],"number":377930}
buildkite-pr-comment-->

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/378700)
* Commit: 77cf258555544ab57371107c5e74213636e55eed

### Failed CI Steps
* [FTR Configs #43](https://buildkite.com/elastic/kibana-pull-request/builds/378700#019b995d-8f49-4de4-8ffd-c76496a63017)
* [FTR Configs #99](https://buildkite.com/elastic/kibana-pull-request/builds/378700#019b995d-8f8b-4524-9ba5-04bc78744c49)
* [FTR Configs #113](https://buildkite.com/elastic/kibana-pull-request/builds/378700#019b996d-18b8-448b-8396-3bd94b7696a7)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/378700#019b995d-8f8b-4524-9ba5-04bc78744c49) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/378700/jobs/019b995d-8f8b-4524-9ba5-04bc78744c49/artifacts/019b9987-d3c0-45a4-a251-41aafd92ab8e) FTR Configs #<span></span>99 / Package policies Package Policy - upgrade when upgrading from an integration package to an input package where no required variable has been added upgrade upgrades the package policy and creates the correct templates
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/378700#019b996d-18b8-448b-8396-3bd94b7696a7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/378700/jobs/019b996d-18b8-448b-8396-3bd94b7696a7/artifacts/019b9998-0418-4206-97e6-7f3d617f7d73) FTR Configs #<span></span>113 / Stateful Observability feature flag testing - Deployment-agnostic Observability Agent Builder API integration tests Observability Agent tool: observability.get_log_change_points when retrieving log change points should detect spike in error logs

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :yellow_heart: [Build #377930](https://buildkite.com/elastic/kibana-pull-request/builds/377930) was flaky cfca8b888e49e7ebb59082822c2d72bbea6ff4d2
* :yellow_heart: [Build #377814](https://buildkite.com/elastic/kibana-pull-request/builds/377814) was flaky aea8c221b9f7c32a991c23e6421b03be110d6f00
* :yellow_heart: [Build #377552](https://buildkite.com/elastic/kibana-pull-request/builds/377552) was flaky 9a230f1b371b2de990575cbd89f6acca52a60287
* :yellow_heart: [Build #377372](https://buildkite.com/elastic/kibana-pull-request/builds/377372) was flaky bab3e681a6ec79ec73eadf5ec23b1d40c560106b

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/378700","number":378700,"commit":"77cf258555544ab57371107c5e74213636e55eed"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377930","number":377930,"commit":"cfca8b888e49e7ebb59082822c2d72bbea6ff4d2"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377814","number":377814,"commit":"aea8c221b9f7c32a991c23e6421b03be110d6f00"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377552","number":377552,"commit":"9a230f1b371b2de990575cbd89f6acca52a60287"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377372","number":377372,"commit":"bab3e681a6ec79ec73eadf5ec23b1d40c560106b"}],"number":378700}
buildkite-pr-comment-->

mattkime commented on x-pack/platform/plugins/private/rollup/public/crud_app/sections/job_list/job_list.test.js:
```diff
@@ -31,69 +28,76 @@ jest.mock('../../services/documentation_links', () => {
   };
 });
 
+jest.mock('./job_table', () => ({
```
ok, thats a good standard. I haven't worked with redux much.

mattkime reviewed (approved):
A couple of things worth considering that I'll let you consider, otherwise overall very good

kapral18 committed:
refactor: extract renderJobCreate helper function
- Extract `renderJobCreate` function into a separate file
- Update test files to use the new `renderJobCreate` helper function

kapral18 commented on x-pack/platform/plugins/private/rollup/public/test/client_integration/job_create_histogram.test.js:
```diff
@@ -5,97 +5,127 @@
   let startMock;
+  let history;
+
+  const renderJobCreate = () => {
```
yes this is big enough to abstract away. Done

kapral18 commented on x-pack/platform/plugins/private/rollup/public/test/client_integration/job_create_histogram.test.js:
```diff
@@ -5,97 +5,127 @@
+  };
+
+  const clickNext = () => fireEvent.click(screen.getByTestId('rollupJobNextButton'));
+  const clickBack = () => fireEvent.click(screen.getByTestId('rollupJobBackButton'));
```
exactly, simplicity over DRY

kapral18 committed:
fix: move button check inside test cases
- Move the check for the button not being disabled inside each test case
- Ensure the button state is verified before performing actions in each test
